cookies 번역 (#381)
Nextjs-kr/Nextjs.kr: docs/02-app/02-api-reference/04-functions/cookies.mdx

---
title: cookies
description: API Reference for the cookies function.
title: 쿠키
description: 쿠키 함수에 대한 API 레퍼런스
related:
  title: Next Steps
  description: For more information on what to do next, we recommend the following sections

>   links:

Reviewer: 삭제된 `links` 다시 추가해주세요!

> `cookies` 기능을 사용하면 [서버 구성요소](/docs/getting-started/react-essentials)에서 HTTP 수신 요청 쿠키를 읽거나 [서버 액션](/docs/app/building-your-application/data-fetching/server-actions) 또는 [라우트 핸들러](/docs/app/building-your-application/routing/router-handlers)에 송신 요청 쿠키를 쓸 수 있습니다.

Reviewer: ```suggestion
`cookies` 기능을 사용하면 [서버 컴포넌트](/docs/getting-started/react-essentials)에서 HTTP 수신 요청 쿠키를 읽거나 [서버 액션](/docs/app/building-your-application/data-fetching/server-actions) 또는 [라우트 핸들러](/docs/app/building-your-application/routing/router-handlers)에 송신 요청 쿠키를 쓸 수 있습니다.
```

> > **참고** :  `cookies()`는 반환된 값을 미리 알 수 없는 [동적 함수](/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-functions)입니다. 레이아웃 또는 페이지에서 사용하면 요청 시 [동적 렌더링](/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-rendering) 경로가 선택됩니다.

Reviewer: ```suggestion
> **참고** :  `cookies()`는 반환된 값을 미리 알 수 없는 **[동적 함수](/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-functions)**입니다. 레이아웃 또는 페이지에서 사용하면 요청 시 경로는 **[동적 렌더링](/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-rendering)**으로 선택됩니다.
```

## `cookies().get(name)`

> 쿠키 이름을 사용하고 이름과 값을 가진 개체를 반환하는 메서드입니다. name이 지정된 쿠키를 찾을 수 없으면 undefined 쿠키가 반환됩니다. 여러 쿠키가 일치하는 경우 첫 번째 일치 항목만 반환됩니다.

Reviewer: ```suggestion
쿠키 이름을 사용하고 이름과 값을 가진 객체를 반환하는 메서드입니다. 쿠키에서 `name`을 찾을 수 없으면 `undefined`가 반환됩니다. 일치하는 쿠키가 여러 개인 경우 가장 먼저 일치하는 쿠키만 반환됩니다.
```

## `cookies().has(name)`

> 쿠키 이름을 사용하고 쿠키의 존재(`true`) 여부(`false`)를 기준으로 `boolean`을 반환하는 메서드입니다.

Reviewer: ```suggestion
쿠키 이름을 사용하고 쿠키의 존재하는지(`true`), 존재하지 않는지(`false`)를 기준으로 `boolean`을 반환하는 메서드입니다.
```

## `cookies().set(name, value, options)`

> 쿠키 이름, 값 및 옵션을 사용하고 나가는 요청 쿠키를 설정하는 메서드입니다.

Reviewer: ```suggestion
쿠키 이름, 값 및 옵션을 받아서 송신 요청 쿠키를 설정하는 메서드입니다.
```

> > **참고** : `.set()`은 [서버 액셕](/docs/app/building-your-application/data-fetching/server-actions) 또는 [라우트 핸들러](/docs/app/building-your-application/routing/router-handlers)에서만 사용할 수 있습니다.

Reviewer: ```suggestion
> **참고** : `.set()`은 [서버 액션](/docs/app/building-your-application/data-fetching/server-actions) 또는 [라우트 핸들러](/docs/app/building-your-application/routing/router-handlers)에서만 사용할 수 있습니다.
```

}
```

## 쿠키 삭제

쿠키를 "삭제"하려면 이름이 같고 값이 비어 있는 새 쿠키를 설정해야 합니다. 쿠키를 즉시 만료하도록 `maxAge`를 `0`으로 설정할 수도 있습니다.

> > 참고 : `.set()`은 [서버 액션](/docs/app/building-your-application/data-fetching/server-actions)  또는 [라우트 핸들러](/docs/app/building-your-application/routing/router-handlers)에서만 사용할 수 있습니다.

Reviewer: ```suggestion
> **참고** : `.set()`은 [서버 액션](/docs/app/building-your-application/data-fetching/server-actions)  또는 [라우트 핸들러](/docs/app/building-your-application/routing/router-handlers)에서만 사용할 수 있습니다.
```
